ci-operator/step-registry/workers-rhel/pre-hook/gcp/workers-rhel-pre-hook-gcp-commands.sh

```diff
@@ -16,13 +16,41 @@ if ! whoami &> /dev/null; then
     fi
 fi
 
-cat > scaleup-pre-hook-gcp.yaml <<- 'EOF'
+if ! test -f "${SHARED_DIR}/proxy_private_url"; then
+  echo "Failed to find bastion proxy info, abort." && exit 1
+fi
+proxy_private_url="$(< ${SHARED_DIR}/proxy_private_url)"
+proxy="http://$(echo ${proxy_private_url} | cut -d@ -f2)"
+proxy_username="$(echo ${proxy_private_url} | cut -d@ -f1 | cut -d\/ -f3 | cut -d: -f1)"
+proxy_password="$(echo ${proxy_private_url} | cut -d@ -f1 | cut -d\/ -f3 | cut -d: -f2)"
+
+cat > scaleup-pre-hook-gcp.yaml << EOF
 - name: Configure RHEL machine on GCP
   hosts: new_workers
   any_errors_fatal: true
-  gather_facts: false
+  gather_facts: true
 
   tasks:
+  - name: insert proxy settings into dnf.conf
+    shell: |
```

Comment:
for adding multiple lines to a configuration file, I guess we can use the `blockinfile` module more easily

Reply:
Let me try, thanks!